console_direct doesnt display stderr directly

CMake warnings are displayed only once the build is finished rather than directly.
e.g. here I added `message(WARNING "NOT using security!")` to the CMakeLists that I expect to be printed during the CMake configure step. But it is not displayed before the end of the make phase.

```
$ colcon build --packages-up-to fastrtps --event-handler console_direct+
Starting >>> fastrtps
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done      
-- Detecting C compile features
-- Detecting C compile features - done       
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features            
-- Detecting CXX compile features - done     
-- Configuring Fast RTPS
-- Version: 1.5.0
-- To change de version modify the file configure.ac
-- Performing Test SUPPORTS_CXX11
-- Performing Test SUPPORTS_CXX11 - Success
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found           
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found              
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- fastcdr library not found...
-- Found asio: /usr/include  
-- Found tinyxml2: /usr/lib/x86_64-linux-gnu/libtinyxml2.so  
-- Looking for pthread.h
-- Looking for pthread.h - found             
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mikael/work/ros2/current_ws/build/fastrtps
Scanning dependencies of target fastrtps     
[  2%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/utils/eClock.cpp.o
[  3%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/log/StdoutConsumer.cpp.o
[  4%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/utils/IPFinder.cpp.o
[  2%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/log/Log.cpp.o
[  5%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/utils/md5.cpp.o
[  6%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/utils/StringMatching.cpp.o
[  7%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/resources/ResourceEvent.cpp.o
[  9%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/resources/TimedEvent.cpp.o
[ 10%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/resources/TimedEventImpl.cpp.o
[ 11%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/resources/AsyncWriterThread.cpp.o
[ 12%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/resources/AsyncInterestTree.cpp.o
[ 13%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/Endpoint.cpp.o
[ 14%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/RTPSWriter.cpp.o
[ 15%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/StatefulWriter.cpp.o
[ 17%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/ReaderProxy.cpp.o
[ 18%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/StatelessWriter.cpp.o
[ 19%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/ReaderLocator.cpp.o
[ 20%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/timedevent/PeriodicHeartbeat.cpp.o
[ 21%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/timedevent/NackResponseDelay.cpp.o
[ 22%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/writer/timedevent/NackSupressionDuration.cpp.o
[ 23%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/history/CacheChangePool.cpp.o
[ 25%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/history/History.cpp.o
[ 26%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/history/WriterHistory.cpp.o
[ 27%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/history/ReaderHistory.cpp.o
[ 28%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/timedevent/HeartbeatResponseDelay.cpp.o
[ 29%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/timedevent/WriterProxyLiveliness.cpp.o
[ 30%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/timedevent/InitialAckNack.cpp.o
[ 31%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/CompoundReaderListener.cpp.o
[ 32%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/WriterProxy.cpp.o
[ 34%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/StatefulReader.cpp.o
[ 35%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/StatelessReader.cpp.o
[ 36%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/RTPSReader.cpp.o
[ 37%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/reader/FragmentedChangePitStop.cpp.o
[ 38%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/messages/CDRMessagePool.cpp.o
[ 39%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/messages/RTPSMessageCreator.cpp.o
[ 40%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/messages/RTPSMessageGroup.cpp.o
[ 42%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/messages/MessageReceiver.cpp.o
[ 43%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/network/NetworkFactory.cpp.o
[ 44%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/network/SenderResource.cpp.o
[ 45%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/network/ReceiverResource.cpp.o
[ 46%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/participant/RTPSParticipant.cpp.o
[ 47%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/participant/RTPSParticipantImpl.cpp.o
[ 48%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/RTPSDomain.cpp.o
[ 50%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/Domain.cpp.o
[ 51%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/participant/Participant.cpp.o
[ 52%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/participant/ParticipantImpl.cpp.o
[ 53%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/publisher/Publisher.cpp.o
[ 54%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/publisher/PublisherImpl.cpp.o
[ 55%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/publisher/PublisherHistory.cpp.o
[ 56%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/subscriber/Subscriber.cpp.o
[ 57%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/subscriber/SubscriberImpl.cpp.o
[ 59%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/subscriber/SubscriberHistory.cpp.o
[ 60%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/transport/UDPv4Transport.cpp.o
[ 61%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/transport/UDPv6Transport.cpp.o
[ 62%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/transport/test_UDPv4Transport.cpp.o
[ 63%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/qos/ParameterList.cpp.o
[ 64%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/qos/ParameterTypes.cpp.o
[ 65%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/qos/QosList.cpp.o
[ 67%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/qos/QosPolicies.cpp.o
[ 68%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/qos/WriterQos.cpp.o
[ 69%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/qos/ReaderQos.cpp.o
[ 70%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/BuiltinProtocols.cpp.o
[ 71%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/participant/PDPSimple.cpp.o
[ 72%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/participant/PDPSimpleListener.cpp.o
[ 73%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/participant/timedevent/RemoteParticipantLeaseDuration.cpp.o
[ 75%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/participant/timedevent/ResendParticipantProxyDataPeriod.cpp.o
[ 76%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/endpoint/EDP.cpp.o
[ 77%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/endpoint/EDPSimple.cpp.o
[ 78%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/endpoint/EDPSimpleListeners.cpp.o
[ 79%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/endpoint/EDPStatic.cpp.o
[ 80%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/liveliness/WLP.cpp.o
[ 81%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/liveliness/WLPListener.cpp.o
[ 82%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/liveliness/timedevent/WLivelinessPeriodicAssertion.cpp.o
[ 84%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/data/ParticipantProxyData.cpp.o
[ 85%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/data/WriterProxyData.cpp.o
[ 86%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/data/ReaderProxyData.cpp.o
[ 87%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/flowcontrol/ThroughputController.cpp.o
[ 88%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/flowcontrol/ThroughputControllerDescriptor.cpp.o
[ 89%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/flowcontrol/FlowController.cpp.o
[ 90%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/exceptions/Exception.cpp.o
[ 92%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/attributes/PropertyPolicy.cpp.o
[ 93%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/common/Token.cpp.o
[ 94%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/xmlparser/XMLParserCommon.cpp.o
[ 95%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/xmlparser/XMLElementParser.cpp.o
[ 96%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/xmlparser/XMLEndpointParser.cpp.o
[ 97%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/xmlparser/XMLProfileManager.cpp.o
[ 98%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/xmlparser/XMLParser.cpp.o
[100%] Linking CXX shared library libfastrtps.so   
src/cpp/CMakeFiles/fastrtps.dir/build.make:2331: recipe for target 'src/cpp/libfastrtps.so.1.5.0' failed
CMakeFiles/Makefile2:85: recipe for target 'src/cpp/CMakeFiles/fastrtps.dir/all' failed
Makefile:149: recipe for target 'all' failed
CMake Warning at CMakeLists.txt:214 (message):
  NOT using security!


/usr/bin/ld: cannot find -lfastcdr
collect2: error: ld returned 1 exit status
make[2]: *** [src/cpp/libfastrtps.so.1.5.0] Error 1
make[1]: *** [src/cpp/CMakeFiles/fastrtps.dir/all] Error 2
make: *** [all] Error 2
--- stderr: fastrtps
CMake Warning at CMakeLists.txt:214 (message):
  NOT using security!


/usr/bin/ld: cannot find -lfastcdr
collect2: error: ld returned 1 exit status
make[2]: *** [src/cpp/libfastrtps.so.1.5.0] Error 1
make[1]: *** [src/cpp/CMakeFiles/fastrtps.dir/all] Error 2
make: *** [all] Error 2
---
Failed   <<< fastrtps	 [ Exited with code 2 ]

Summary: 0 packages finished [28.46s]
  1 package failed: fastrtps
  1 package had stderr output: fastrtps
```

--- a/colcon_core/event_handler/console_direct.py
+++ b/colcon_core/event_handler/console_direct.py
@@ -43,3 +43,4 @@ class ConsoleDirectEventHandler(EventHandlerExtensionPoint):
                 sys.stderr.buffer.write(data.line)
             else:
                 sys.stderr.write(data.line)
+            sys.stderr.flush()

